src/main/kotlin/com/openlattice/assembler/processors/AddMembersToOrganizationAssemblyProcessor.kt

```diff
@@ -43,21 +42,18 @@ data class AddMembersToOrganizationAssemblyProcessor(
         ReadOnly {
 
     @Transient
-    private lateinit var acm: AssemblerConnectionManager
+    private var acm: AssemblerConnectionManager? = null
 
     override fun process(entry: MutableMap.MutableEntry<UUID, OrganizationAssembly?>): Void? {
         val organizationId = entry.key
         val assembly = entry.value
-        if (assembly == null) {
-            throw IllegalStateException("Encountered null assembly while trying to add new principals " +
-                    "${authorizedPropertyTypesOfEntitySetsByPrincipal.keys} to organization $organizationId.")
-        } else {
-            val dbName = PostgresDatabases.buildOrganizationDatabaseName(organizationId)
-            acm.connect(dbName)
-                    .use { dataSource ->
-                        acm.addMembersToOrganization(dbName, dataSource, authorizedPropertyTypesOfEntitySetsByPrincipal)
-                    }
-        }
+        assembly ?: throw IllegalStateException("Encountered null assembly while trying to add new principals " +
```

Comment:
consider check( assembly != null ) { "..." }

Reply:
why is that better? or just for re-use purpose?

Reply:
Idiomatic style for checking state using kotlin std lib.